Add docs for service accounts 8.5 (#46801)
grafana/grafana: docs/sources/administration/service-accounts/create-service-account.md

---
title: Create a service account in Grafana
menuTitle: Create a service account
description: 'How to create a service account in Grafana'
weight: 50
keywords:
  - Service accounts
---

# Create a service account in Grafana

> A service account is a user account that can be used to run automated or compute workloads. For more information, refer to [About service accounts]({{< relref "../service-accounts/about-service-accounts.md#">}}).

Reviewer: ```suggestion
A service account is a user account that you can use to run automated or compute workloads. For more information about how you can use service accounts, refer to [About service accounts]({{< relref "../service-accounts/about-service-accounts.md#">}}).
```

For more information about creating service accounts via the API, refer to [Create service account via API]({{< relref "../../http_api/serviceaccount.md#create-service-account">}}).

Before you begin:

- Ensure you have added the feature toggle for service accounts `service-accounts`. [Enable service accounts]({{< relref "./enable-service-accounts.md#">}})

> - Ensure you have permission to create and edit service accounts, for more information refer to [About users and permissions]({{< relref "../manage-users-and-permissions/about-users-and-permissions.md#">}})

Reviewer: ```suggestion
- Ensure you have permission to create and edit service accounts. For more information about user permissions, refer to [About users and permissions]({{< relref "../manage-users-and-permissions/about-users-and-permissions.md#">}}).
```

**To create a service account:**

1. Sign in to Grafana and hover your cursor over the organization icon in the sidebar.
1. Click **Service accounts**.
1. Click **New service account**.

> 1. Enter a display name.

Reviewer 2: The formatting is broken here. I'd recommend to split Display Name docs to a separate section to be honest, there are constraints and patterns we suggest, so it's worth to  have it separate. What do you think?

<img width="1095" alt="Screenshot 2022-04-07 at 12 43 00" src="https://user-images.githubusercontent.com/1861190/162181717-2137e473-d189-4bf5-890c-23ca0253e9eb.png">

> 1. Enter a display name.

Reviewer: Is display name a field? If so, then I would go with:

Enter a **Display name**.

> - This name needs to be unique to start with, as it determines the id.

Reviewer: ```suggestion
  - The display name must be unique as it determines the ID associated with the service account.
```

> - We recommend that you use a consistent naming convention when you name service accounts. A consistent naming convention can help you scale and maintain service accounts in the future.

Reviewer: ```suggestion
  - We recommend that you use a consistent naming convention when you name service accounts. A consistent naming convention can help you scale and maintain service accounts in the future.
```

> - Do not worry about the naming you can change this name later.

Reviewer: ```suggestion
- You can change the display name at any time.
```

> ---

Reviewer 2: I am not sure if there is any advise against this, but I think it would be good to include couple of screenshots to make is visually clear how to do it. For example like [we did it here](https://grafana.com/docs/grafana/latest/enterprise/access-control/manage-role-assignments/manage-user-role-assignments/).

Reviewer: I am generally not in favor of adding screenshots unless it adds clarity that cannot otherwise be communicated with the text. From what I've seen, the UI is very simple, so I'm not sure if adding a screenshot adds any real value. It also adds to tech debt when the UI changes (which it will, at some point). In the manage users link that you reference, that screenshot existed previously so I kept it there when I refactored the content. In that example, I should really remove that image. (noting down on to-do list!)

title: Create a service account in Grafana
menuTitle: Create a service account
aliases: [docs/sources/administration/service-accounts/add-service-account-token.md]
description: 'How to create a service account in Grafana'
weight: 50
keywords:
  - Service accounts
---

# Create a service account in Grafana

> You can create a service account in the service account tab of the organization. To manage your computed workload with a user that is not tied to a user specifically.

Reviewer: Remove the first sentence because it'll be covered in the task steps. Add more of an intro here about what a service account is. "A service account is a ..... that enables you to ...'. For more information about service accounts, refer to xxx."

> Consider the following restriction when you create a service account:

Reviewer: ```suggestion
Consider the following restrictions when you create a service account:
```

> - A service account is identified by its name, which is unique to the entire suite of organizations. The name is set upon creation of the service account

Reviewer: The second sentence starts to get into the task steps. I would remove it, doesn't seem necessary. I think what you are saying is "When you create a service account, you enter a name." - in which case, I'd remove it.

Reviewer: What is a 'suite of organizations'? Do you mean organizations within an instance of Grafana?

Author: i think we have mentioned both of these things but in different places. Specifically when running into the first step when naming.

> - Service accounts live on a organizational level and are restricted as such

Reviewer: Find another word for 'live'

Reviewer 2: What about "Service accounts are mapped to an organization where they are created"?

For more information about service accounts refer to [About service accounts in Grafana]({{< relref "./about-service-accounts.md">}}).

For more information about creating service accounts via the API, refer to [Create service account via API]({{< relref "../../http_api/serviceaccount.md#create-service-account">}}).

Before you begin:

> - Ensure you have added the feature toggle for service accounts `service-accounts`. Refer to [Enable service accounts]({{< relref "./enable-service-accounts.md#">}})

Reviewer: Adjust wording based on previous feedback.

> - Ensure you have permission to create and edit service accounts, for more information refer to [About users and permissions]({{< relref "../manage-users-and-permissions/about-users-and-permissions.md#">}})

Reviewer: same here.

**To create a service account:**

1. Hover your mouse over the organization icon in the sidebar.
1. Click service accounts. Grafana opens the service accounts tab.
1. Click **New service account**.

> 1. Enter a display name, this will be unique within the organization

Reviewer: When you want to provide extra information about a step, format it like this:

1. Enter a display name. 
    The display name is unique within an organization.

2. blah blah blah.
     You are now able to blah blah blah.

> 1. Enter a display name, this will be unique within the organization

Author: you can always change it, but at least we can mention this
